### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by Jia-A**
This pull request contains product page, contexts and filter for my e-commerce website.

### CODE REVIEW

File: src/dot-components/Product-listing/product.jsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                    <div class="main-info-cont">
+                        <h3 class="info-head">{item.title}</h3>
+                        <p class="sm-txt">{item.author}</p>
+                        <p className="sub-info">{item.ratings}<span><i class="fas fa-star"></i></span></p>
+                        <p class="sub-info">Rs. {item.price}</p>
+                    </div>
+                    {/* <div class="sub-info-cont">
+                        
+                        
+                    </div> */}
```
Comment (shuklahimanshu81) [inline src/dot-components/Product-listing/product.jsx:109]: remove comment and extra space

File: src/dot-components/Product-listing/product.css [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    font-weight: 500;
+}
+
+.primary-btn-border{
+    background-color: transparent;
+}
+/* .main-info-cont{
+    padding: 10px;
+    margin: 10px;
+} */
```
Comment (shuklahimanshu81) [inline src/dot-components/Product-listing/product.css]: Here too remove comments.

**Review by shuklahimanshu81 — APPROVED** [review]
Everything is good, just try to extract the functions in different files and then using them will make it more better .

File: src/Context/cartContext.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            return {...cartState,
+               cart: [
+                  ...cartState.cart, 
+                  {...action.payload, quantity: 1}]}
+
+                  case 'Increase_quantity':
+                    return {...cartState,
+                       cart: 
+                        cartState.cart.map((item) => item.id === action.payload.id ? 
+                            {...item, quantity: item.quantity + 1} : item)
```
Comment (tanishk15sharma) [inline src/Context/cartContext.js]: I think you dont need to `increase` or `decrease` explicitly . you can use backend for this .

File: src/Context/filterContext.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import axios from "axios";
+
+const initVal = {
+    categoryFil: [],
+    priceRangeFil: 500,
+    ratingFil: 1,
+    sortByPriceFil: "",
+    searchFil: "",
+    showProduct: [],
+    realProduct: []
```
Comment (tanishk15sharma) [inline src/Context/filterContext.js:18]: Suggestion: You can give more specific name .

File: src/dot-components/Product-listing/product.css [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    align-self: center;
+}
+.info-head{
+    font-weight: 500;
+}
+
+.primary-btn-border{
+    background-color: transparent;
+}
+/* .main-info-cont{
```
Comment (tanishk15sharma) [inline src/dot-components/Product-listing/product.css:70]: you can remove this snippet

**Review by tanishk15sharma — APPROVED** [review]
Code is working and immutable . suggested  some changes to make it more simple .

File: src/Context/filterContext.js (hunk 4 above)
Comment (Jia-A) [reply src/Context/filterContext.js:18]: Okay will do that.

File: src/Context/cartContext.js (hunk 3 above)
Comment (Jia-A) [reply src/Context/cartContext.js]: Oh okay, will try to change that.

**Pull request merged**